lib/network_ssl/network_ssl.c

```diff
@@ -346,6 +354,12 @@ dowrite(struct network_ssl_ctx * ssl)
 			    &ssl->write_callback));
 	}
 
+	/* Restore the old SIGPIPE handler. */
```

Comment:
We need to do this if we do the callback at line 352.

Also, it's possible for OpenSSL to write to sockets while performing a read operation -- for example, if the first thing you try to do with a connection is reading from it, since OpenSSL will perform the TLS handshake at that point.